components/signup/signup_email/index.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
+
+import {GenericAction, ActionFunc} from 'mattermost-redux/types/actions';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {ServerError} from 'mattermost-redux/types/errors';
+
+import {createUser} from 'mattermost-redux/actions/users';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getTeamInviteInfo} from 'mattermost-redux/actions/teams';
+
+import {setGlobalItem} from 'actions/storage';
+import {loginById} from 'actions/views/login';
+import {getPasswordConfig} from 'utils/utils.jsx';
+
+import {GlobalState} from '../../../types/store';
+
+import SignupEmail from './signup_email';
+
+export type PasswordConfig = {
+    minimumLength: number;
+    requireLowercase: boolean;
+    requireNumber: boolean;
+    requireSymbol: boolean;
+    requireUppercase: boolean;
+}
+
+export type Props = {
+    location: {search: string};
+    enableSignUpWithEmail: boolean;
+    siteName: string;
+    termsOfServiceLink?: string;
+    privacyPolicyLink?: string;
+    customDescriptionText?: string;
+    passwordConfig?: PasswordConfig;
+    hasAccounts: boolean;
+    actions: Actions;
+};
+
+export type State = {
+    loading: boolean;
+    inviteId?: string;
+    token?: string;
+    email?: string;
+    teamName?: string;
+    noOpenServerError?: boolean;
+    isSubmitting?: boolean;
+    nameError?: React.ReactNode;
+    emailError?: React.ReactNode;
+    passwordError?: React.ReactNode;
+    serverError?: React.ReactNode;
+};
+
+type TeamInviteInfo = {
+    display_name: string;
+    description: string;
+    name: string;
+    id: string;
+};
+
+export type Actions = {
+    createUser: (user: UserProfile, token: string, inviteId: string, redirect: string) => Promise<{data: UserProfile} | {error: ServerError}>;
+    loginById: (id: string, password: string, mfaToken?: string) => Promise<{data: boolean} | {error: ServerError}>;
+    setGlobalItem: (name: string, value: string) => {data: boolean};
+    getTeamInviteInfo: (inviteId: string) => Promise<{data: TeamInviteInfo} | {error: ServerError}>;
+};
+
+function mapStateToProps(state: GlobalState) {
+    const config = getConfig(state);
+
+    const enableSignUpWithEmail = config.EnableSignUpWithEmail === 'true';
+    const siteName = config.SiteName as string;
```

Comment:
Since `config.SiteName` is nullable, it might be better to define this as `config.SiteName || ''` instead of using a type assertion since that could cause problems elsewhere if someone were to check something like `siteName.length`

Reply:
I am 5/5 on this too as to avoid type casting.